### PR TITLE
New protoLibrary option for the Gradle plugin

### PR DESCRIPTION
**Pull request by swankjesse**
Configuring a project as a protoLibrary will cause the generated
.jar file to include .proto sources.

Also automatically add a dependency when a protoPath or protoSource
depends on a project.

Also make protoPath and protoSource dependencies transitive by default.

### CODE REVIEW

File: wire-library/docs/wire_compiler.md [hunk 1]
```diff
@@ -292,6 +292,21 @@ wire {
...
 ```
 
+### Proto Libraries
+
+By default, `.proto` input files are not included in the generated `.jar` artifact. Use the
+`protoLibrary` option to include them:
+
+```groovy
+wire {
+  protoLibrary = true
```
Comment (oldergod) [inline wire-library/docs/wire_compiler.md:302]: Great idea

File: wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt [hunk 2]
```diff
@@ -90,6 +95,17 @@ class WirePlugin : Plugin<Project> {
       it.description = "Aggregation task which runs every generation task for every given source"
     }
 
+    if (extension.protoLibrary) {
+      val libraryProtoSources = File("${buildDir}/wire/proto-sources")
+      val sourceSets = project.extensions.getByType(SourceSetContainer::class.java)
+      sourceSets.getByName("main") { main: SourceSet ->
+        main.resources.srcDir(libraryProtoSources)
+      }
```
Comment (oldergod) [inline wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt:103]: I wonder if that would work on Android with variants, but I guess we don't expect Android modules to be proto libraries anyway.
I can check later to see what happens. We could just log a warning if `main` isn't found.

**Review by oldergod — APPROVED** [review]
Super nice thank you

**Comment by jrodbx** [comment]
nice!

**Pull request merged**